src/commands/createSwaConfigFile.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { writeFile } from 'fs-extra';
+import { join } from 'path';
+import { EntryInfo, promise } from 'readdirp';
+import { TextDocument, Uri, window, workspace } from 'vscode';
+import { ext } from 'vscode-azureappservice/out/src/extensionVariables';
+import { IActionContext } from 'vscode-azureextensionui';
+import { configFileName } from '../constants';
+import { localize } from '../utils/localize';
+import { getWorkspaceFolder } from '../utils/workspaceUtils';
+
+export async function createSwaConfigFile(context: IActionContext): Promise<void> {
+    const workspacePath: string = (await getWorkspaceFolder(context)).uri.fsPath;
+    let configFilePath: string | undefined = await getExistingConfigFilePath(workspacePath);
+
+    if (configFilePath) {
+        void ext.ui.showWarningMessage(localize('swaConfigFileExists', 'Static Web App configuration file "{0}" already exists.', configFilePath));
+    } else {
+        configFilePath = join(workspacePath, configFileName);
+        await writeFile(configFilePath, exampleConfigFile);
+    }
+
+    const textDocument: TextDocument = await workspace.openTextDocument(Uri.file(configFilePath));
+    await window.showTextDocument(textDocument);
+}
+
+async function getExistingConfigFilePath(workspacePath: string): Promise<string | undefined> {
+    const entries: EntryInfo[] = await promise(workspacePath, { fileFilter: configFileName });
```

Comment:
Why not just use fs-extra ala https://github.com/microsoft/vscode-azurefunctions/blob/85bed3eac6a8a236a5d2d886fafb6269a61c44ec/src/funcConfig/local.settings.ts#L73

Reply:
Here's what the [docs](https://docs.microsoft.com/en-us/azure/static-web-apps/configuration#file-location) say about the config file's location:

> The recommended location for the staticwebapp.config.json is in the folder set as the app_location in the workflow file. However, the file may be placed in any location within your application source code folder.

I think that means it could live in any arbitrary nested folder so I made the search recursive with `readdirp`

Reply:
Hmm, in general, we probably don't want to do a recursive search through their entire project.  If they have a relatively large project without a swa config file, then it could take a while.  I think it makes sense to look maybe 2 folders deep.

I know Eric decided on one folder deep for Function projects: https://github.com/microsoft/vscode-azurefunctions/blob/85bed3eac6a8a236a5d2d886fafb6269a61c44ec/src/commands/createNewProject/verifyIsProject.ts#L22-L56

We also might be able to use their .yml config file (if they have it) to put the file in the app_location folder.

Reply:
Well one reason I'm not doing a full recursive search is because that code gets called a lot, including when the extension is activated. The scenario here is a command the user explicitly started, so you may or may not want to do something different. Also the code you linked is pretty old in Functions. At this point, I'd probably recommend the `vscode.workspace.findFiles` API over pure `fs-extra`

Reply:
I updated the search to use `vscode.workspace.findFiles` and removed the `readdirp` dependency